ComSemApp/student/views.py

```diff
@@ -702,7 +695,13 @@ def get_context_data(self, **kwargs) -> dict[str, Any]:
         context : dict[str, Any] = super(SpeakingPracticeView, self).get_context_data(**kwargs)
         context['problems'] : list[dict[str, str | float]] = {}
         context['student'] : int = self.student.id
-        
+
+        session : SpeakingPracticeSession = SpeakingPracticeSession.objects.filter(student=self.student).latest('date')
```

Comment:
The issue I was having is here. If you have no sessions, (which I didn't) an error gets thrown here. I fixed it with a try-except but there may be a better way using django built in functions perhaps

Reply:
The try-catch seems to be the standard way of doing this. The exception was thrown by the `latest()` call, and the [official documentation](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#latest) discusses its use.